kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/KafkaProtocolFilterReconciler.java

```diff
@@ -0,0 +1,210 @@
+                                                                                                                   Class<R> secondaryClass,
+                                                                                                                   Function<SecureConfigInterpolator.InterpolationResult, Stream<String>> resourceNameExtractor) {
+        return InformerEventSourceConfiguration.from(
+                secondaryClass,
+                KafkaProtocolFilter.class)
+                .withPrimaryToSecondaryMapper((KafkaProtocolFilter filter) -> {
+                    Object configTemplate = filter.getSpec().getConfigTemplate();
+                    var interpolationResult = secureConfigInterpolator.interpolate(configTemplate);
+                    Set<ResourceID> resourceIds = resourceNameExtractor.apply(interpolationResult)
+                            .map(name -> new ResourceID(name, ResourcesUtil.namespace(filter)))
+                            .collect(Collectors.toSet());
+                    LOGGER.debug("Filter {} references {}(s) {}", ResourcesUtil.name(filter), secondaryClass.getName(), resourceIds);
+                    return resourceIds;
+                })
+                .withSecondaryToPrimaryMapper(secret -> {
+                    Set<ResourceID> resourceIds = ResourcesUtil.filteredResourceIdsInSameNamespace(
+                            context,
+                            secret,
+                            KafkaProtocolFilter.class,
+                            filter -> {
+                                Object configTemplate = filter.getSpec().getConfigTemplate();
+                                var interpolationResult = secureConfigInterpolator.interpolate(configTemplate);
+                                return resourceNameExtractor.apply(interpolationResult)
+                                        .anyMatch(secretNameFromVolume -> secretNameFromVolume.equals(ResourcesUtil.name(secret)));
+                            });
+                    LOGGER.debug("{} {} referenced by Filters {}", secondaryClass.getName(), ResourcesUtil.name(secret), resourceIds);
+                    return resourceIds;
+                })
+                .build();
+    }
+
+    @Override
+    public UpdateControl<KafkaProtocolFilter> reconcile(
+                                                        KafkaProtocolFilter filter,
+                                                        Context<KafkaProtocolFilter> context) {
+
+        var now = ZonedDateTime.ofInstant(clock.instant(), ZoneId.of("Z"));
+
+        ConditionBuilder conditionBuilder = newResolvedRefsCondition(filter, now);
+
+        var extentSecrets = context.getSecondaryResourcesAsStream(Secret.class)
+                .map(ResourcesUtil::name)
+                .collect(Collectors.toSet());
+        LOGGER.debug("Extent secrets: {}", extentSecrets);
```

Comment:
Might be just me, but I found the use of the word "extent" slightly obscure.

Reply:
Can't have helped that it's spelled wrong. I meant [extant](https://dictionary.cambridge.org/dictionary/english/extant). But `existing` would be better, so I'll change it.